werkzeug/test.py

```diff
@@ -338,6 +340,27 @@ def __init__(self, path='/', base_url=None, query_string=None,
                         self._add_file_from_data(key, value)
                     else:
                         self.form.setlistdefault(key).append(value)
+        if json:
+            input_count = 0
```

Comment:
I agree with the reduction, and I'll make the change, but I put the logic in the `if json:` block because I'm worried that changing the exception message for multiple input sources would be a breaking change for some folks. What do you think, @untitaker?

Also, thanks for checking out the PR over the weekend.